packages/gatsby/src/commands/develop-process.ts

```diff
@@ -108,34 +128,114 @@ module.exports = async (program: IProgram): Promise<void> => {
     throw e
   }
 
-  // Start bootstrap process.
-  const { gatsbyNodeGraphQLFunction, workerPool } = await bootstrap({ program })
-
-  // Start the createPages hot reloader.
-  bootstrapPageHotReloader(gatsbyNodeGraphQLFunction)
-
-  // Start the schema hot reloader.
-  bootstrapSchemaHotReloader()
+  const app = express()
 
-  const { queryIds } = await calculateDirtyQueries({ store })
+  const developConfig: MachineConfig<IBuildContext, any, AnyEventObject> = {
+    id: `build`,
+    initial: `initializing`,
+    states: {
+      initializing: {
+        invoke: {
+          src: `initialize`,
+          onDone: {
+            target: `initializingDataLayer`,
+            actions: `assignStoreAndWorkerPool`,
+          },
+        },
+      },
+      initializingDataLayer: {
+        invoke: {
+          src: `initializeDataLayer`,
+          data: ({ parentSpan, store }: IBuildContext): IDataLayerContext => {
+            return { parentSpan, store, firstRun: true }
+          },
+          onDone: {
+            actions: `assignDataLayer`,
+            target: `doingEverythingElse`,
+          },
+        },
+      },
+      doingEverythingElse: {
+        invoke: {
+          src: async ({
+            gatsbyNodeGraphQLFunction,
+            graphqlRunner,
+            workerPool,
+            store,
+            app,
+          }): Promise<void> => {
+            // All the stuff that's not in the state machine yet
+
+            // These were previously in `bootstrap()` but are now
+            // in part of the state machine that hasn't been added yet
+            await postBootstrap({ parentSpan: bootstrapSpan })
```

Comment:
this moves `postBootstrap` before `rebuildSchemaWithSitePage`, `extractQueries`, `writeOutRedirects` - currently we have it like this - https://github.com/gatsbyjs/gatsby/blob/8e4bc927a91c4e73e06eafc69e0673c7ba0553d4/packages/gatsby/src/bootstrap/index.ts#L52-L60

Is this re-order needed for incoming changes? Ideally we don't swap order if we don't have to

Reply:
Good spot. I move this into a state machine in the next PR, but I'll switch it around here and double check I've got it in the right place in the new one